src/main.cpp

```diff
@@ -1,7 +1,32 @@
 #include <csignal>
+#include <list>
+#include <sys/types.h>
+#include <sys/wait.h>
 #include <spdlog/spdlog.h>
 #include "client.hpp"
 
+sig_atomic_t is_inserting_pid = false;
```

Comment:
Should be replaced with a mutex, no?

Reply:
I forgot signals can be handled by any thread, so yeah it's not safe atm. But I'm pretty sure using a mutex in a signal handler is unsafe so we can't do that. Apparently a good way handle signals in multithreaded programs is to have a dedicated thread that handles signals like [here](https://www.linuxjournal.com/article/2121), [here](https://devarea.com/linux-handling-signals-in-a-multithreaded-application/), and [here](http://poincare.matf.bg.ac.rs/~ivana/courses/ps/sistemi_knjige/pomocno/apue/APUE/0201433079/ch12lev1sec8.html). That way we can properly use mutexes. So I will try that